src/dkg/round2.rs

```diff
@@ -310,19 +313,25 @@ impl CombinedPublicPackage {
         buf
     }
 
-    pub fn serialize_into<W: io::Write>(&self, mut writer: W) -> io::Result<()> {
+    pub fn serialize_into<W: io::Write>(&self, mut writer: W) -> Result<(), IronfishFrostError> {
         let sender_identity = &self.packages[0].sender_identity;
         sender_identity.serialize_into(&mut writer)?;
-        write_variable_length(writer, &self.packages, |writer, pkg| {
-            pkg.serialize_without_sender_into(writer)
-        })
+        Ok(write_variable_length(
+            writer,
+            &self.packages,
+            |writer, pkg| {
+                pkg.serialize_without_sender_into(writer)
+                    .map_err(|_| io::Error::other("serialize_into failed"))
+            },
+        )?)
     }
 
-    pub fn deserialize_from<R: io::Read>(mut reader: R) -> io::Result<Self> {
+    pub fn deserialize_from<R: io::Read>(mut reader: R) -> Result<Self, IronfishFrostError> {
         let sender_identity = Identity::deserialize_from(&mut reader)?;
 
         let packages = read_variable_length(reader, move |reader| {
             PublicPackage::deserialize_without_sender_from(reader, sender_identity.clone())
+                .map_err(|_| io::Error::other("deserialization failed"))
```

Comment:
I wonder if it's better to have `read_variable_length not rely on an `io::Result` but instead rely on `Result<T, IronfishFrostError>` since that change is being made in a lot of places here - or if maybe we want to stick with io::Results in a lot of these places. Unclear to me yet.